packages/block-editor/src/components/media-upload-progress/index.native.js

```diff
@@ -154,8 +168,16 @@ export class MediaUploadProgress extends Component {
 					retryMessage,
 				} ) }
 			</View>
-		);
-	}
+		),
+		[
+			isUploadInProgress,
+			isUploadFailed,
+			progress,
+			progressBarStyle,
+			renderContent,
+			props,
```

Comment:
Sharing similar concerns about passing all `props` to the dependency array, I'm worried it could lead to some unwanted side effects or unnecessary renders. Could we review the dependencies that are being used here to ensure they're all necessary?